### PR TITLE
[ADD] mail_split_by_partner_conf: Allow configuration for sending only one email to all recipients

### PR DESCRIPTION
**Pull request by phucngta**
Default behavior of Odoo is to send one email for each person.
The objective of this module would be to allow configuring if the native behavior should be applied or if only one email should be sent to all recipients (convenient to know who else has received the email).

============================
This would be a selection "Split mail by recipient partner" in the Mail Templates form after the field "Auto Delete": [(default,Project Default), (split,One mail for each recipient (Odoo's default)), (merge,One mail for all recipients)]

In mail template, default value will be default.
Project Default means that we will get the value defined on a config parameter "default_mail_split_by_partner_conf".
At installation create the Configuration Paramter with value to split and if ir.configuration has been deleted use split as it is in the native behavior of Odoo.
In Mail Templates and in Mail Mail, add a field selection "Split mail by recipient partner" so can know how this email has been processed.

### CODE REVIEW

File: mail_split_by_partner_conf/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa - <antonio.espinosa@tecnativa.com>
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/tests/__init__.py]: ?

File: mail_split_by_partner_conf/__init__.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# © 2017 Phuc.nt - <phuc.nt@komit-consulting.com> - <phuc.nt@komit-consulting.com>
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/__init__.py]: 2 times your email

File: mail_split_by_partner_conf/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from . import models
+from . import wizard
+from . import tests
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/__init__.py]: Should not import tests

File: mail_split_by_partner_conf/models/mail_mail.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+from email.utils import formataddr
+
+from odoo import models, fields, api
+
+
+class MailMail(models.Model):
+    _inherit = 'mail.mail'
+
+    # The number of mail is created to notify followers
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/models/mail_mail.py]: The number of mails

File: mail_split_by_partner_conf/models/mail_mail.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    mail_count = fields.Integer(string='Number of mails')
+
+    split_mail_by_recipients = fields.Selection([
+        ('split', 'One mail for each recipient'),
+        ('merge', 'One mail for all recipients'),
+        ('default', 'Project Default')],
+        string='Split mail by recipient partner',
+        default='default',
+        help='',
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/models/mail_mail.py]: Remove empty help or add one :)

I think you can remove.

File: mail_split_by_partner_conf/models/mail_mail.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+from email.utils import formataddr
+
+from odoo import models, fields, api
+
+
+class MailMail(models.Model):
+    _inherit = 'mail.mail'
+
+    # The number of mail is created to notify followers
+    mail_count = fields.Integer(string='Number of mails')
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/models/mail_mail.py]: I don't understand why you need this extra field, I don't see where it is used.

File: mail_split_by_partner_conf/models/mail_thread.py [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+from odoo import models, api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    @api.multi
+    def message_get_recipient_values(self, notif_message=None,
+                                     recipient_ids=None):
+        # merge followers when mail
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/models/mail_thread.py]: It seems you did not finish the sentence in your comment.

File: mail_split_by_partner_conf/security/ir.model.access.csv [hunk 8]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_mail_split_by_partner_conf_mail_split_by_partner_conf,mail_split_by_partner_conf.mail_split_by_partner_conf,model_mail_split_by_partner_conf_mail_split_by_partner_conf,,1,0,0,0
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/security/ir.model.access.csv]: Do we really need this? How is it used?

File: mail_split_by_partner_conf/wizard/mail_compose_message.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/wizard/mail_compose_message.py]: Missing copyright

File: mail_split_by_partner_conf/wizard/mail_compose_message.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+from odoo import models, api
+
+
+class MailComposer(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    @api.model
+    def generate_email_for_composer(self, template_id, res_ids, fields=None):
+        # TODO - is need check case fields is none
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/wizard/mail_compose_message.py]: Could you check now? Or add a explicit message in the section Know Issues of the readme?

File: mail_split_by_partner_conf/models/mail_template.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+class MailTemplate(models.Model):
+    _inherit = 'mail.template'
+
+    split_mail_by_recipients = fields.Selection([
+        ('split', 'One mail for each recipient'),
+        ('merge', 'One mail for all recipients'),
+        ('default', 'Project Default')],
+        string='Split mail by recipient partner',
+        default='default',
+        help='',
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/models/mail_template.py]: Should remove empty help message

File: mail_split_by_partner_conf/README.rst [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+* Nguyen Tan Phuc <phuc.nt@komit-consulting.com>
+
+Do not contact contributors directly about support or help with technical issues.
+
+Funders
+-------
+
+The development of this module has been financially supported by:
+
+* Komit-consulting <https://komit-consulting.com>
```
Comment (jcdrubay) [inline mail_split_by_partner_conf/README.rst]: Komit <https://komit-consulting.com>

**Review by jcdrubay — CHANGES_REQUESTED** [review]

File: mail_split_by_partner_conf/wizard/mail_compose_message.py (hunk 10 above)
Comment (phucngta) [reply mail_split_by_partner_conf/wizard/mail_compose_message.py]: I have done but forgot to remove it

File: mail_split_by_partner_conf/security/ir.model.access.csv (hunk 8 above)
Comment (phucngta) [reply mail_split_by_partner_conf/security/ir.model.access.csv]: we don't need this, it was generated by scaffold. Let me delete it

File: mail_split_by_partner_conf/models/mail_mail.py [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+        ('merge', 'One mail for all recipients'),
+        ('default', 'Project Default')],
+        string='Split mail by recipient partner',
+        default='default',
+    )
+
+    @api.multi
+    def send_get_mail_to(self, partner=None):
+        # format address if partner is a list
+        if len(partner) > 1:
```
Comment (novawish) [inline mail_split_by_partner_conf/models/mail_mail.py]: Must check if `partner` is `None` or not, otherwise it will raise `Error`

**Review by novawish — CHANGES_REQUESTED** [review]

File: mail_split_by_partner_conf/models/mail_template.py [hunk 14]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# © 2017 Phuc.nt - <phuc.nt@komit-consulting.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields
```
Comment (hieulucky111) [inline mail_split_by_partner_conf/models/mail_template.py:5]: Order import? https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

File: mail_split_by_partner_conf/models/mail_thread.py [hunk 15]
```diff
@@ -0,0 +1,18 @@
+# -*- coding: utf-8 -*-
+# © 2017 Phuc.nt - <phuc.nt@komit-consulting.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
```
Comment (hieulucky111) [inline mail_split_by_partner_conf/models/mail_thread.py:5]: Order import? https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

File: mail_split_by_partner_conf/wizard/mail_compose_message.py [hunk 16]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# © 2017 Phuc.nt - <phuc.nt@komit-consulting.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
```
Comment (hieulucky111) [inline mail_split_by_partner_conf/wizard/mail_compose_message.py:5]: Order Import https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

**Review by hieulucky111 — APPROVED** [review]
Some minor adjustments. LGTM over all.

**Pull request closed without merging**